src/create-http-client.ts

```diff
@@ -127,6 +129,20 @@ export default function createHttpClient(
       ...newParams,
     })
   }
+
+  if (config.onBeforeRequest) {
```

Comment:
Does the order of registering the handlers matter?

Reply:
Yes, ideally the order shouldn't change. Do you want me to write a unit test for this?

Reply:
A comment that explains why the order is important is fine.

Reply:
done